### PR TITLE
Python: Fix Agent Framework tool parameter defaults

### PR DESCRIPTION
**Pull request by ump45nose**
### Motivation and Context

`KernelFunction.as_agent_framework_tool()` currently builds a field with a parameter default and then immediately overwrites it with a required field. As a result, optional Semantic Kernel parameters become required in the Agent Framework tool schema.

### Description

- preserve every optional field by using `is_required` as the source of truth, including explicit `None` defaults
- add a focused regression test that verifies both `default=5` and `default=None` remain optional

Validation:

- `uv run pytest tests/unit/functions/test_kernel_function_from_method.py -q` — 37 passed
- `uv run ruff check semantic_kernel/functions/kernel_function.py tests/unit/functions/test_kernel_function_from_method.py` — passed
- `uv run ruff format --check semantic_kernel/functions/kernel_function.py tests/unit/functions/test_kernel_function_from_method.py` — passed
- `git diff --check` — passed

### Contribution Checklist

- [x] I added a focused regression test for the corrected behavior
- [x] I reviewed the change against the SK Contribution Guidelines
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: python/semantic_kernel/functions/kernel_function.py [hunk 1]
```diff
@@ -452,7 +452,8 @@ def as_agent_framework_tool(
                         param.type_,
                         Field(description=param.description, default=param.default_value),
                     )
-                fields[param.name] = (param.type_, Field(description=param.description))
+                else:
+                    fields[param.name] = (param.type_, Field(description=param.description))
         input_model = create_model("InputModel", **fields)  # type: ignore
```
[the remark this answers is not in the document]
Comment (ump45nose) [reply python/semantic_kernel/functions/kernel_function.py:459]: Fixed in `3ee4563be`. Field construction now keys off `param.is_required`; optional parameters always pass their declared default to Pydantic, including an explicit `None`, while required parameters remain required.

File: python/tests/unit/functions/test_kernel_function_from_method.py [hunk 2]
```diff
@@ -140,6 +140,25 @@ def decorated_function() -> None:
...
+    def function(count: int = 5) -> int:
+        return count
+
+    native_function = KernelFunction.from_method(method=function, plugin_name="MockPlugin")
+    tool = native_function.as_agent_framework_tool(kernel=Mock())
+    field = tool.input_model.model_fields["count"]
+
+    assert field.default == 5
+    assert not field.is_required()
+
```
[the remark this answers is not in the document]
Comment (ump45nose) [reply python/tests/unit/functions/test_kernel_function_from_method.py:163]: Covered in `3ee4563be`. The regression now includes both `count: int = 5` and `label: str | None = None`, and asserts that both generated fields are optional while preserving their respective defaults.

**Comment by ump45nose** [comment]
已将分支变基到当前 `upstream/main` 并安全推送（`f58bb39a3`）。

本地验证：
- `uv run pytest tests/unit/functions/test_kernel_function_from_method.py`：37 passed（1 个既有 Pydantic 弃用警告）
- `git diff --check upstream/main...HEAD`：通过

新提交触发的 Python、.NET、拼写与合并门禁工作流目前均为 `action_required`，看起来需要维护者批准 fork 的工作流后才能执行。烦请批准运行并复审。

该分支的变基、测试与说明由 AI 辅助完成。
